### DISCUSSION TITLE
Ability to specify a different auth header?

### DISCUSSION THREAD

**Discussion by derekantrican** [coddingtonbear/obsidian-local-rest-api#60] [Ideas]
I'm looking to use this on my home server and put it behind nginx as a reverse proxy (I already use nginx to expose a bunch of other services on my local network). On nginx, I have defined a username & password so I can auth with it as `Authorization: Basic [base64 username & password]`. But nginx does not support multiple Authorization headers, so it looks like I can't use this plugin behind nginx.

I would like the ability, in the plugin settings, to specify my own header (say `X-API-KEY`) that I can pass the API key on, instead of the `Authorization: Bearer [key]`

**Comment by coddingtonbear** [comment]
For those who find this discussion after-the-fact; this capability was added via 371b9512453c3920f6aa3ac57925abdf0ec98445 and will be released in the next release of this plugin.

**Reply by derekantrican** [reply]
Looks like the new version was released! Thanks!

I'm trying to get it configured. Looks like the openapi doc page (at least the "Try it out" actions) don't use the customized header name (still use "Authorization"). How easy would that be to fix?

**Reply by coddingtonbear** [reply]
I have no idea, but I'm afraid I'm also not up for modifying the docs for things we're exposing under "Advanced Settings"; good luck if you decide to pursue that, though!
